src/core/lib/transport/error_utils.cc

```diff
@@ -52,7 +53,19 @@ void grpc_error_get_status(grpc_error* error, grpc_millis deadline,
   if (GPR_LIKELY(error == GRPC_ERROR_NONE)) {
     if (code != nullptr) *code = GRPC_STATUS_OK;
     if (slice != nullptr) {
-      grpc_error_get_str(error, GRPC_ERROR_STR_GRPC_MESSAGE, slice);
+      // Normally, we call grpc_error_get_str(
+      //   error, GRPC_ERROR_STR_GRPC_MESSAGE, slice).
+      // We can fastpath since we know that:
+      // 1) Error is null
+      // 2) which == GRPC_ERROR_STR_GRPC_MESSAGE
+      // 3) The resulting slice is statically known.
+      // 4) Said resulting slice is of length 0 ("").
+      // This means 3 movs, instead of 10s of instructions and a strlen.
+      static const char* no_err = "";
+      slice->refcount = &grpc_core::NoopRefcount;
```

Comment:
Can we call `grpc_slice_from_static_buffer()` here instead of inlining its contents?  Or would that defeat the purpose of the optimization?

Reply:
It would add extra instructions, yes.

An alternate: grpc_slice_from_static_buffer_internal, defined inline in slice_internal.h. The non-internal method just calls it. Internal callers (mostly hpack_parser.cc) save the unnecessary call/ret.

Reply:
Fixed; added grpc_slice_from_static_buffer/string_internal().